tests/capture/workflow/test_capture_qnode.py

```diff
@@ -347,6 +351,150 @@ def circuit(x):
     assert list(out.keys()) == ["a", "b"]
 
 
+@pytest.mark.parametrize(
+    "dev_name", ["default.qubit", pytest.param("lightning.qubit", marks=pytest.mark.skip)]
+)
+class TestDevicePreprocessing:
+    """Unit tests for preprocessing and executing qnodes with program capture."""
+
+    def test_non_native_ops_execution(self, dev_name):
+        """Test that operators that aren't natively supported by a device can be executed by a qnode."""
+        dev = qml.device(dev_name, wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            # QFT not supported on DQ or LQ
+            qml.QFT(wires=[0, 1])
+            return qml.state()
+
+        assert qml.math.allclose(circuit(), [0.5] * 4)
+
+    @pytest.mark.parametrize("mcm_method", [None, "deferred"])
+    @pytest.mark.parametrize("shots", [None, 1000])
+    def test_mcms_execution_deferred(self, dev_name, mcm_method, shots):
+        """Test that defer_measurements is reflected in the execution results of a device."""
+        # Parametrized over mcm_method because default (None) method is "deferred"
+        # Shots tests should check shape and ~vaguely~ validate distribution. Use the following test for inspiration:
+        # https://github.com/PennyLaneAI/pennylane-lightning/blob/589392d0dd31737e559048626698911439bc2706/tests/test_eval_jaxpr.py#L782-L783
+        # Use postselection with DQ, not with LQ
+        # Use reset with both devices
+
+        dev = qml.device(dev_name, wires=3, shots=shots)
+        postselect = 1 if dev_name == "default.qubit" else None
+
+        @qml.qnode(dev, mcm_method=mcm_method)
+        def circuit():
+            qml.Hadamard(0)
+            qml.CNOT([0, 1])  # |Φ⁺⟩ = (1/√2) (|00⟩ + |11⟩)
+            qml.measure(0, reset=True, postselect=postselect)
+            return {
+                "expval": (qml.expval(qml.Z(0)), qml.expval(qml.Z(1))),
+                "state": qml.sample(wires=[0, 1]) if shots else None,
```

Comment:
```suggestion
                "samples": qml.sample(wires=[0, 1]) if shots else None,
```
Given why I see "state", I think `qml.state()`.

Reply:
Oops, lol I was using `qml.state()` before and forgot to change it. Thanks!